### PR TITLE
HomeWork4 String

### PR DESCRIPTION
**Pull request by qpuhuk**
HW6 String

### CODE REVIEW

File: src/main/java/AlexLink/HomeWork/String/TaskAdd.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//
+//1) Найти в строке не только запятые, но и другие знаки препинания.
+//Подсчитать общее их количество.
+
+
+import java.util.Arrays;
+
+public class TaskAdd {
+    public static void main(String[] args) {
+        final String SENTENCE = "Ехал я по городу и услышал топот! Обернулся, а там стая лошадей. " +
```
Comment (svetamolchanova) [inline src/main/java/AlexLink/HomeWork/String/TaskAdd.java]: we use uppercase for static final variable. see Code Convension

Comment (qpuhuk) [reply src/main/java/AlexLink/HomeWork/String/TaskAdd.java]: fixed it

File: src/main/java/AlexLink/HomeWork/String/TextFormatter.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package AlexLink.HomeWork.String;
+
+import java.util.Arrays;
+
+public class TextFormatter {
+    private static String pattern = "\\s|([.,!?-]\\s)|\\.";
```
Comment (atlamax) [inline src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: private static final String PUNCTUATION_MARKS_PATTERN = ....

File: src/main/java/AlexLink/HomeWork/String/TextFormatter.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    protected static int countWordsInString(String str) {
+        int count = 0;
+        String[] del = str.split(pattern);
+        for (String s : del) {
```
Comment (atlamax) [inline src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: del.length ?

File: src/main/java/AlexLink/HomeWork/String/Task4.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    public static void main(String[] args) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String pattern = "\\.";
+        String[] divSentence = SENTENCE.split(pattern);
+        for (String temp : divSentence) {
+            String format = temp.trim();
+            int countWord = TextFormatter.countWordsInString(format);
+            boolean pol = TextFormatter.checkPolindromInSentence(format);
+            if ((countWord > 3 && countWord <= 5) || pol)
```
Comment (atlamax) [inline src/main/java/AlexLink/HomeWork/String/Task4.java]: 1) countWord >=3 ?
2) Если предложение подходит по количеству слов, то **не надо** проверять его на палиндром
```
if ((countWord >= 3 && countWord <= 5) || TextFormatter.checkPolindromInSentence(format)) {
    stringBuilder.append(format).append(". ");
}
```

File: src/main/java/AlexLink/HomeWork/String/TextFormatter.java (hunk 2 above)
Comment (qpuhuk) [reply src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: fixed it

File: src/main/java/AlexLink/HomeWork/String/TextFormatter.java (hunk 3 above)
Comment (qpuhuk) [reply src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: fixed it

File: src/main/java/AlexLink/HomeWork/String/Task4.java (hunk 4 above)
Comment (qpuhuk) [reply src/main/java/AlexLink/HomeWork/String/Task4.java]: fixed it

File: src/main/java/AlexLink/HomeWork/String/TextFormatter.java (hunk 2 above)
Comment (atlamax) [reply src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: @qpuhuk You didn't push your changes

Comment (qpuhuk) [reply src/main/java/AlexLink/HomeWork/String/TextFormatter.java]: All done. I'm not finished it at that moment. Thanks a lot

**Pull request merged**